src/CommonLib/SRWLockWrapper.cpp

```diff
@@ -0,0 +1,13 @@
+#include "stdafx.h"
+#include "SRWLockWrapper.h"
+
+SRWLockWrapper::SRWLockWrapper(const SRWLOCK& lock)
+    : m_lock(lock)
+{
+    AcquireSRWLockExclusive(const_cast<SRWLOCK*>(&m_lock));
```

Comment:
It seems to be excessive to be downcasting to SRWLOCK just so the call to SRWLockWrapper accepts const. I would drop it here.

Reply:
what's the benefit of this wrapper? it introduce additional overhead by just wrapping Windows API. Why should we waste CPU?

Reply:
The benefit is that you don't need to think about releasing resources in methods and is a widely used practice in C++ (https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) that simplifies code.

The overhead of this wrapper is minimal if any at all, compiler knows now to inline constructor and method calls so I'm pretty sure generated code wouldn't change at all:

![image](https://user-images.githubusercontent.com/1697911/38529469-b75e5730-3c19-11e8-942a-731d1ec379f0.png)

Reply:
I ran  AcquireSRWLockExclusive() &  ReleaseSRWLockExclusive() vs. SRWLockWrapper lock(srwLock) million times, the first takes 15 ticks while the other takes  140 ticks.

Reply:
after applying that, the perf is same. Compiler does put the method inline :)